nepalingo-web/src/components/user_auth/User_auth.tsx

```diff
@@ -0,0 +1,156 @@
+import React, { useState } from "react";
+import supabase from "./supabaseClient";
+import { useNavigate } from "react-router-dom";
+
+const User_auth: React.FC = () => {
+  const [action, setAction] = useState<"Sign Up" | "Log In">("Sign Up");
+  const [username, setUsername] = useState("");
```

Comment:
instead of `""` you can leave it empty. That way you know you need to handle the case where it is undefined.

Reply:
when I tried that, I got this error:
 
 Type 'undefined' is not assignable to type 'string'.
 
 what to do now?

Reply:
we talked about leaving this for now